### PR TITLE
fix(snapshotting): adjust PVC size calculation with 110% buffer

### PR DESCRIPTION
**Pull request by jean0215**
**Problem:**

When downloading an image, if the model size is close to the PVC size, it may result in no space.

**Solution:**

Apply 110% buffer to PVC size calculation to account for potential overhead and ensure minimum 1GB allocation. Also optimize file deletion in S3 download by performing it before downloads to free up space.

**Related Issue:**

**Test plan:**

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved calculation of persistent volume claim (PVC) sizes by adding a 10% buffer and ensuring a minimum allocation of 1Gi, preventing insufficient storage issues.
  * Enhanced incremental download process by deleting obsolete local files before downloading new or updated files, optimizing local storage usage and synchronization reliability.

* **Tests**
  * Added comprehensive tests for PVC size calculation, covering edge cases and error handling to ensure accurate storage sizing.

* **Other**
  * Added debug logging to provide clearer insights during storage allocation and file synchronization operations.

### CODE REVIEW

File: pkg/controller/master/common/snapshotting/manager.go [hunk 1]
```diff
@@ -512,26 +512,31 @@ func (m *Manager) ensureServiceAccountAndRoleBinding(namespace string) error {
...
+	// Using integer arithmetic to avoid floating point precision issues
+	bufferedSize := (size * 11) / 10
+	
+	// Convert to GB and round up
+	divisor := int64(1024 * 1024 * 1024)
+	gbSize := bufferedSize / divisor
+	if bufferedSize%divisor > 0 {
+		// Round up if there's any remainder
+		gbSize++
+	}
```
[the remark this answers is not in the document]
Comment (guangbochen) [reply pkg/controller/master/common/snapshotting/manager.go:531]: @jean0215 this seems to be reasonable, please help to update it.

**Review by guangbochen — APPROVED** [review]
